src/rules/noRedundantJsdocRule.ts

```diff
@@ -79,6 +79,7 @@ function walk(ctx: Lint.WalkContext): void {
                 break;
 
             case ts.SyntaxKind.JSDocClassTag:
+            case ts.SyntaxKind.JSDocThisTag:
```

Comment:
Although I'm inclined to believe you that the rule is breaking on this case, there should be some test changes to validate that this change fixes the issue (and doesn't cause new ones). #3414 is a good one to go off of.

Reply:
Three test cases come to mind:

* `@this`
* `@this {}`
* `@this {blah}`